### DISCUSSION TITLE
"Access-Control-Allow-Origin: *" header and CORS in fastapi

### DISCUSSION THREAD

**Discussion by aryapunni** [fastapi/fastapi#8576] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from fastapi.middleware.cors import CORSMiddleware

app = FastAPI()

origins = [
    "http://localhost.tiangolo.com",
    "https://localhost.tiangolo.com",
    "http://localhost",
    "http://localhost:8080",
]

app.add_middleware(
    CORSMiddleware,
    allow_origins=origins,
    allow_credentials=True,
    allow_methods=["*"],
    allow_headers=["*"],
)


@app.get("/")
async def main():
    return {"message": "Hello World"}
```

### Description

Hello, 
I am using fastapi for my server. I am trying to bring in data from different servers to mine using their APIs. One of them requires CORS and I have setup CORS as it is given in the fastapi documentation. The server from which I am collecting data requires:
HTTP/1.1 200 OK\r\n
Access-Control-Allow-Origin: *\r\n
Content-Length: 10\r\n
\r\n
ServerTime,DataTime,Optional -- this as the response. From the current setup I am getting date, server, content length, and content type as the response. How can I add "Access-Control-Allow-Origin: *\r\n" as response? Should I add it as a response header separately? 
Thank you
Arya

### Operating System

Linux

### FastAPI Version

0.63.0

### Python Version

3.8.10

**Comment by upeshphable** [comment]
@aryapunni CORS needs to be enabled on the other server since you are consuming their APIs to bring data. Your IP/Hostname should be in their allowed origins.

**Comment by aryapunni** [comment]
Hi @upeshphable, Thank you. The other server just dumps data into my server on regular intervals when I set up a path in its settings. Their requirements says that the host server should enable CORS and  "Access-Control-Allow-Origin: *\r\n" this should be there in the response headers.Thanks

**Comment by AlrasheedA** [comment, marked as answer]
Hi @aryapunni, "*" is a wildcard allowing all origins, it can be added instead of the other origins you have in your example. See [documentation](https://fastapi.tiangolo.com/tutorial/cors/#wildcards) for more details. Keep in mind, **it is better to specify explicitly the allowed origins**. 

Consult with the service provider that you are integrating with, and ask them about alternatives for the wildcard. Also, don't forget about adding [authentication](https://fastapi.tiangolo.com/tutorial/security/#security-intro), if it is not already there.

**Comment by aryapunni** [comment]
Hi @AlrasheedA 
Thank you. I will consult service provider for more information.
